src/stream/src/executor/mview/materialize.rs

```diff
@@ -418,13 +437,88 @@ impl<S: StateStore> MaterializeExecutor<S, BasicSerde> {
             conflict_behavior,
             version_column_index: None,
             is_dummy_table: false,
+            toastable_column_indices: None,
             may_have_downstream: true,
             depended_subscription_ids: HashSet::new(),
             metrics,
         }
     }
 }
 
+/// Debezium's default unavailable value placeholder for TOAST columns.
+const DEBEZIUM_UNAVAILABLE_VALUE: &str = "__debezium_unavailable_value";
+
+/// Fast string comparison to check if a string equals `DEBEZIUM_UNAVAILABLE_VALUE`.
+/// Optimized by checking length first to avoid expensive string comparison.
+fn is_unavailable_value_str(s: &str) -> bool {
+    s.len() == DEBEZIUM_UNAVAILABLE_VALUE.len() && s == DEBEZIUM_UNAVAILABLE_VALUE
+}
+
+/// Check if a datum represents Debezium's unavailable value placeholder.
+/// This function handles both scalar types and one-dimensional arrays.
+fn is_debezium_unavailable_value(
+    datum: &Option<risingwave_common::types::ScalarRefImpl<'_>>,
+) -> bool {
+    match datum {
+        Some(risingwave_common::types::ScalarRefImpl::Utf8(val)) => is_unavailable_value_str(val),
+        Some(risingwave_common::types::ScalarRefImpl::Jsonb(jsonb_ref)) => {
+            // For jsonb type, check if it's a string containing the unavailable value
+            jsonb_ref
+                .as_str()
+                .map(is_unavailable_value_str)
+                .unwrap_or(false)
+        }
+        Some(risingwave_common::types::ScalarRefImpl::Bytea(bytea)) => {
+            // For bytea type, we need to check if it contains the string bytes of DEBEZIUM_UNAVAILABLE_VALUE
+            // This is because when processing bytea from Debezium, we convert the base64-encoded string
+            // to `DEBEZIUM_UNAVAILABLE_VALUE` in the json.rs parser to maintain consistency
+            if let Ok(bytea_str) = std::str::from_utf8(bytea) {
+                is_unavailable_value_str(bytea_str)
+            } else {
+                false
+            }
+        }
+        Some(risingwave_common::types::ScalarRefImpl::List(list_ref)) => {
+            // For list type, check if it contains exactly one element with the unavailable value
+            // This is because when any element in an array triggers TOAST, Debezium treats the entire
+            // array as unchanged and sends a placeholder array with only one element
```

Comment:
what if some elements change in the list? Debezium resends the whole list or like `["__debezium_unavailable_value", "new value", "__debezium_unavailable_value"]`

Reply:
Debezium will resends the whole list, since this column is updated.